junit-platform-commons/src/main/java/org/junit/platform/commons/util/KotlinReflectionUtils.java

```diff
@@ -65,6 +74,44 @@ public static boolean isKotlinSuspendingFunction(Method method) {
 		return false;
 	}
 
+	/**
+	 * Determines whether the supplied class is a {@code DefaultImpls} class
+	 * generated by the Kotlin compiler.
+	 *
+	 * <p>See
+	 * <a href="https://kotlinlang.org/docs/interfaces.html#jvm-default-method-generation-for-interface-functions">Kotlin documentation</a>
+	 * for details.
+	 *
+	 * @since 5.13.3
+	 */
+	@API(status = INTERNAL, since = "5.13.3")
+	public static boolean isKotlinInterfaceDefaultImplsClass(Class<?> clazz) {
+		if (!isKotlinType(clazz) || !DEFAULT_IMPLS_CLASS_NAME.equals(clazz.getSimpleName()) || !isStatic(clazz)) {
+			return false;
+		}
+
+		Class<?> enclosingClass = clazz.getEnclosingClass();
+		if (enclosingClass != null && enclosingClass.isInterface()) {
+			return Arrays.stream(clazz.getDeclaredMethods()) //
+					.anyMatch(method -> isCompilerGeneratedDefaultMethod(method, enclosingClass));
+		}
+
+		return false;
+	}
+
+	private static boolean isCompilerGeneratedDefaultMethod(Method method, Class<?> enclosingClass) {
+		if (isStatic(method) && method.getParameterCount() > 0) {
+			var parameterTypes = method.getParameterTypes();
+			if (parameterTypes[0] == enclosingClass) {
+				var originalParameterTypes = parameterTypes.length == 1 //
+						? EMPTY_CLASS_ARRAY //
+						: Arrays.stream(parameterTypes).skip(1).toArray(Class[]::new);
```

Comment:
For use cases like this one, I'd typically go with `System.arraycopy()` instead of a `Stream`.

Reply:
Refactored in 74d8da3fb42c5dc8115d4c2d05bdad968e53e3ee